### ISSUE TITLE
mkvtoolnix-32bit-7.9.0 Error

### ISSUE THREAD

**Issue by b2kguga** [mbunkus/mkvtoolnix#1223]
Hi guys

there is an error on the newer version mkvtoolnix-32bit-7.9.0. There is a missing Api in advapi32.dll. On WinXP the app is linking to RegGetValueA Api that does not exists on Advapi32.dll (On WinXP 32 bits).

The RegGetValueA works only in WinXP 64 bits and not for 32 bits version.

For winXP32 and older, it must use SHRegGetValue (in shlwapi.dll for WinXp SP2 ) or  RegQueryValueEx. Can you please fix it ?

All of the ".exe" files that uses this api seems to be from a function that retrieves info from the registry at "Software\Microsoft\Windows\CurrentVersion\App Paths\mmg.exe". 

Sources: http://stackoverflow.com/questions/5140374/visual-c-err-reggetvaluea-could-not-be-located
https://msdn.microsoft.com/en-us/library/windows/desktop/ms724868%28v=vs.85%29.aspx

The function that uses this call to HKEY as a verificaton (Returning TRUE or FALSE) can be replaced by something like:

Proc FixPointer::
    Local @pcbData
    Uses edx, edi, esi, ecx

    mov D@pcbData 0
    lea eax D@pcbData
    call 'shlwapi.SHRegGetValueA' &HKEY_LOCAL_MACHINE, {"Software\Microsoft\Windows\CurrentVersion\App Paths\mmg.exe",0},
                                  &NULL, 2, &NULL, &NULL, eax
    If eax = &ERROR_SUCCESS
        mov eax &TRUE
    Else
        xor eax eax
    End_If
    
EndP

**Comment by b2kguga** [comment]
Btw...Found it on the source. The code is inside windows.cpp.
The fix can be done replacing this:

bool is_installed() {
  auto sub_key  = "Software\\Microsoft\\Windows\\CurrentVersion\\App Paths\\mmg.exe";
  auto data_len = DWORD{};

  if (ERROR_SUCCESS != RegGetValueA(HKEY_LOCAL_MACHINE, sub_key, NULL, RRF_RT_REG_SZ, NULL, NULL, &data_len))
    return false;

  return data_len > 1;
}
}}

With this:


bool
is_installed() {
  auto sub_key  = "Software\\Microsoft\\Windows\\CurrentVersion\\App Paths\\mmg.exe";
  auto data_len = DWORD{};

  if (ERROR_SUCCESS != SHRegGetValueA(HKEY_LOCAL_MACHINE, sub_key, NULL, RRF_RT_REG_SZ, NULL, NULL, &data_len))
    return false;

  return data_len > 1;
}

}}

**Comment by mbunkus** [comment]
I don't support Windows XP anymore. See the list of [supported operating systems and versions](/mbunkus/mkvtoolnix/wiki/Officially-supported-operating-systems-and-versions).

That being said: a user has submitted a patch that allegedly restores XP compatibility. It will be part of the next release. But like I said: I don't support XP, and if MKVToolNix breaks for some other reason then that's that.

**Comment by b2kguga** [comment]
Thanks for answering. I made a fix for this. I built a dll from where all exe calls the corresponding function to winxp. Since on version 7.9.0 he only problem is this Api, i simply replaced it and so far everything works as expected. (At least concerning this particular problem). Here is the link of the patches for this version and also the  new dll built for it. Just replace the exe and ll inside the rar file with the ones in the main directory and it should work. (others bugs i didn´t analysed it).
Too bad you don´t support XP anymore. There are still lots of people that uses it and also loves mkvtoolnix. https://mega.co.nz/#!aohk2AAa!I6hHRBhGTAcppZUbcrS3ablg24EJ8_uweRK5_zpwMts
